Source/src/modules/ModuleSceneManager.cpp

```diff
@@ -158,4 +163,17 @@ void Hachiko::ModuleSceneManager::SwitchTo(const char* file_path)
 {
     scene_ready_to_load = true;
     scene_to_load = file_path;
-}
\ No newline at end of file
+    currentScenePath = file_path;
+}
+
+void Hachiko::ModuleSceneManager::ReloadScene()
+{
+    if (std::filesystem::exists(currentScenePath))
+    {
+        LoadScene(currentScenePath.c_str());
+    }
+    else
+    {
+        CreateEmptyScene();
+    }
+}
```

Comment:
Do we really need this? Can't we just do ModuleSceneManager::SwitchTo() on the same scene path and hold current_scene_path outside this class? Or if we really want to do this, don't you think instead of creating an empty scene would mean losing all unsaved data? Maybe if the scene doesn't exist in the assets we can have something like:
1. Autosave current scene to an arbitrary scene file
2. Load that scene file.

Reply:
As I see, not sure because I didn't do ModuleSceneManager::SwitchTo(), there is a diference between that function and ModuleSceneManager::ReloadScene(). On the first case, we want to switch to another scene, from level 1 to level 2 for example, so maybe we want to maintain the health of the player with the same value. Meanwhile, on the second one, we want to reload scene as it was at first, changing some values. That's why I thought that it should be different functions.

Regarding losing all unsaved data, you are right but if you want to reload the scene I think it's because you don't want to save the changes in the scene. If you want to save it, you can do it through File button. I don't know why we want to save it if someone clicks the reload button. We can talk about it class.